pkg/controller/operands/networkAddons_test.go

```diff
@@ -362,6 +363,59 @@ var _ = Describe("CNA Operand", func() {
 			Expect(req.Conditions).To(BeEmpty())
 		})
 
+		type ovsAnnotationParams struct {
+			annotationExists  bool
+			annotationValue   string
+			ovsDeployExpected bool
+		}
+		table.DescribeTable("when reconciling ovs-cni", func(o ovsAnnotationParams) {
+			hcoOVSConfig := commonTestUtils.NewHco()
+			hcoOVSConfig.Annotations = map[string]string{}
+
+			if o.annotationExists {
+				hcoOVSConfig.Annotations["deployOVS"] = o.annotationValue
+			}
+
+			existingResource := NewNetworkAddons(hcoOVSConfig)
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
```

Comment:
is this the normal flow? `NewNetworkAddons()` and only then `handler.ensure(req)` is called?
IIUC, `handler.ensure()` is calling the `updateCr()` function, which updates the annotation if it does not exist.
If that is so, then should the annotation not exist - it'll get added too late when `updateCr()` is called. if this is true, we may need to find a more suitable place the update the annotation if it does not exist.

Reply:
After @orenc1 explained that indeed the flow where the annotation will be added in a certain reconcile and handled in the next one is by design, I'm ok with this.